src/chi-vue/src/components/data-table/DataTable.tsx

```diff
@@ -1286,21 +1286,142 @@ export default class DataTable extends Vue {
+    const row: JSX.Element[] = [];
+    const rowCells: JSX.Element[] = [];
+    const subLevelContent = [];
+
+    if (this._expandable && bodyRow.nestedContent) {
+      subLevelContent.push(
+        this._printSublevelContent(bodyRow.nestedContent, rowLevel === 'child' ? 'child' : 'parent')
+      );
+    }
+    bodyRow.data.forEach((rowCell: any, index: number) => {
+      let cellData: any;
+      if (!!rowCell.template && !!this.$scopedSlots[rowCell.template]) {
+        if (typeof rowCell === 'object' && rowCell.payload) {
+          cellData = (this.$scopedSlots[rowCell.template] as NormalizedScopedSlot)(rowCell.payload);
+        }
+      } else if (typeof rowCell === 'object' && !!rowCell.value) {
+        cellData = rowCell.value;
+      } else if (typeof rowCell === 'string' || typeof rowCell === 'number') {
+        cellData = rowCell;
+      } else {
+        cellData = null;
+      }
+      if (index === 0 && rowLevel === 'grandChild') {
+        rowCells.push(<td class={`${UTILITY_CLASSES.PADDING.LEFT[6]}`}>{cellData}</td>);
+      } else if (index === 0 && rowLevel === 'child') {
+        rowCells.push(<td class={`${UTILITY_CLASSES.PADDING.LEFT[4]}`}>{cellData}</td>);
+      } else {
+        rowCells.push(<td>{cellData}</td>);
+      }
+    });
+    row.push((<tr>{rowCells}</tr>) as JSX.Element[] & JSX.Element);
+    if (bodyRow.nestedContent) {
+      row.push(subLevelContent as JSX.Element[] & JSX.Element);
+    }
+    return row;
+  }
+
+  _printSublevelContent(sublevelData: DataTableRowNestedContent, contentLevel: 'parent' | 'child') {
+    if (sublevelData.template) {
+      const template = (this.$scopedSlots[sublevelData.template] as NormalizedScopedSlot)(sublevelData.payload);
+
+      return (
+        <tr>
+          <td colspan={Object.keys(this.data.head).length} class={`${UTILITY_CLASSES.PADDING.LEFT[4]}`}>
+            {template}
+          </td>
+        </tr>
+      );
+    } else if (sublevelData.table && sublevelData.table.data) {
+      return sublevelData.table.data.map((bodyRow: DataTableRow) => {
+        return this._printRow(bodyRow, contentLevel === 'child' ? 'grandChild' : 'child');
+      });
+    }
+    return (
+      <tr>
+        <td colspan={Object.keys(this.data.head).length} class={`${UTILITY_CLASSES.PADDING.LEFT[4]}`}>
+          {sublevelData.value}
+        </td>
+      </tr>
+    );
+  }
+
   render() {
     const classes = this._dataTableClasses(this.config.style, this._sortable),
       head = this._head(),
       toolbar = this._toolbar(),
       bulkActions = this._bulkActions(),
       body = this._body(),
-      pagination = this._pagination();
+      pagination = this._pagination(),
+      printHead = this._printHead(),
+      printBody = this._printBody(),
+      printFooter = this._printFooter();
 
     return (
       <div class={classes} role="table" ref="dataTable" data-table-number={dataTableNumber}>
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/table_role
Can you check if `<table>` inside an element with `role="table"` is valid?

Reply:
It doesn't explicitly say, but based on this statement "A relevant use case for the ARIA table role is when CSS's display property overrides the native semantics of a table, such as by display: grid. In this case, you can use the ARIA table roles to re-add the semantics." I'd say it's valid 🤔